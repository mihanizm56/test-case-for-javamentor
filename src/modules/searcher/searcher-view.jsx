import React, { memo } from 'react'
import MenuItem from '@material-ui/core/MenuItem'
import { Button, Select, LangList } from '../../components'

export const SearcherView = memo(
    ({ handleSearchClick, handleSearchChange, listData, options, selectedValue }) => (
        <div className='searcher-wrapper'>
            <div className='searcher-container'>
                <div className='searcher__select'>
                    <Select handleChange={handleSearchChange} selectedValue={selectedValue}>
                        {options.map((option, index) => (
                            <MenuItem className='searcher_option' value={option.value} key={index}>
                                {option.value}
                            </MenuItem>
                        ))}
                    </Select>
                </div>
                <Button handleClick={handleSearchClick} />
            </div>
            <div className='searcher-list'>
                <LangList listData={listData} />
            </div>
        </div>
    )
)
