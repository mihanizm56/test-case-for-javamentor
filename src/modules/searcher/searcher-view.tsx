import MenuItem from "@material-ui/core/MenuItem";
import React, { memo } from "react";
import { Button, LangList, Select } from "../../components";
import { OptionsType, SearcherViewProps } from "./types";

export const SearcherView = memo(
    ({ handleSearchClick, handleSearchChange, langData, options, selectedValue }: SearcherViewProps) => (
        <div className="searcher-wrapper">
            <div className="searcher-container">
                <div className="searcher__select">
                    <Select handleChange={handleSearchChange} selectedValue={selectedValue}>
                         {options.map((option: OptionsType, index: number): React.ReactNode => (
                            <MenuItem className="searcher_option" value={option.value} key={index}>
                                 {option.value}
                            </MenuItem>
                       ))}
                    </Select>
                </div>
                <Button handleClick={handleSearchClick} />
            </div>
            <LangList langData={langData} />
        </div>
    ),
);
