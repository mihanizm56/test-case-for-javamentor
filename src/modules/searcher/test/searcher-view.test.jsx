import React from "react";
import { shallow } from "enzyme";
import { SearcherView } from "../searcher-view";

describe("test SearcherView", () => {
  const searcherViewProps = {
    handleSearchClick: jest.fn(),
    handleSearchChange: jest.fn(),
    langData: [
      {
        title: "test_title",
        createdAt: 1,
        numberOfProjects: 1,
        link: "test_link",
        logo: "test_logo"
      }
    ],
    options: [{ value: "test-option" }],
    selectedValue: "test_value"
  };
  const component = shallow(<SearcherView {...searcherViewProps} />);

  describe("test SearcherView", () => {
    it("test whole render with props", () => {
      expect(component).toMatchSnapshot();
    });
  });
});
