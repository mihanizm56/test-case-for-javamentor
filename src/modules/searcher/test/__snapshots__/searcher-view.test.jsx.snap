// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SearcherView test SearcherView test whole render with props 1`] = `
<div
  className="searcher-wrapper"
>
  <div
    className="searcher-container"
  >
    <div
      className="searcher__select"
    >
      <Memo()
        handleChange={[MockFunction]}
        selectedValue="test_value"
      >
        <WithStyles(MenuItem)
          className="searcher_option"
          key="0"
          value="test-option"
        >
          test-option
        </WithStyles(MenuItem)>
      </Memo()>
    </div>
    <Memo()
      handleClick={[MockFunction]}
    />
  </div>
  <Memo()
    langData={
      Array [
        Object {
          "createdAt": 1,
          "link": "test_link",
          "logo": "test_logo",
          "numberOfProjects": 1,
          "title": "test_title",
        },
      ]
    }
  />
</div>
`;
